src/Tasks/Microsoft.NET.Build.Tasks/GenerateGlobalUsings.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class GenerateGlobalUsings : TaskBase
+    {
+        [Required]
+        public ITaskItem[] Usings { get; set; }
+
+        [Required]
+        public ITaskItem OutputFile { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            if (Usings.Length == 0)
+            {
+                return;
+            }
+
+            var usings = Usings.Select(UsingInfo.Read)
+                .OrderBy(static k => k, UsingInfoComparer.Instance)
+                .Distinct(UsingInfoComparer.Instance);
+
+            StringBuilder builder = new();
+            builder.AppendLine("// <auto-generated/>");
+            foreach (var @using in usings)
+            {
+                builder.Append("global using ");
+
+                if (@using.Static)
+                {
+                    builder.Append("static ");
+                }
+
+                if (!string.IsNullOrEmpty(@using.Alias))
+                {
+                    builder.Append(@using.Alias)
+                        .Append(" = ");
+                }
+
+                builder.Append("global::")
+                    .Append(@using.Namespace)
```

Comment:
If the namespace name contains C# reserved keywords, then is the `Using` item expected to contain the @ escapes already (thus making it specific to C#), or should the task add them?

Reply:
I want to treat the `Include` as an opaque C# string. I really don't want to add language rules to the SDK